tests/Unit/CrudPanel/CrudPanelColumnsTest.php

```diff
@@ -127,16 +127,17 @@ class CrudPanelColumnsTest extends BaseDBCrudPanelTest
             'relation_type' => 'HasOne',
         ],
         'accountDetails__nickname' => [
-            'name'        => 'accountDetails.nickname',
+            'name'        => 'accountDetails',
             'label'       => 'AccountDetails.nickname',
             'type'        => 'relationship',
             'key'         => 'accountDetails__nickname',
             'priority'    => 1,
+            'attribute' => 'nickname',
             'tableColumn' => false,
             'orderable'   => false,
             'searchLogic' => false,
             'relation_type' => 'HasOne',
-            'entity' => 'accountDetails.nickname',
+            'entity' => 'accountDetails',
             'model' => 'Backpack\CRUD\Tests\Unit\Models\AccountDetails',
         ],
```

Comment:
Hmmm what about the old test, with `accountDetails.nickname`? Does that stop working?

Reply:
Indeed. That's what we are fixing here. 

That test was added when I added the code in `ColumnsProtectedMethods.php` that I am refactoring now. 

I added that way because of the `key` that was generated later so I kept the `attribute` in the `name` and `entity`. 

That lead to other unexpected consequences thus I now refactored `key` to be generated first and remove the actual attribute from the relation string, as it technically is not part of the relation string anyway, it's just what we want to show.